docs/adr/adr_001_cache_bust.md

```diff
@@ -0,0 +1,322 @@
+
+Similarly, when a single collection is synchronized manually (or in the Rust client), we first fetch the list of latest changes, lookup the collection timestamp, and then fetch the changeset with the appropriate `?_expected` value.
+
+The timestamps provided are the collection records timestamp, not the collections metadata timestamp. In other words, the timestamp won't be bumped if only the collection metadata is modified, and not its records data.
+
+The collections content signatures rely on certificates that are relatively short-lived. Since some collections are not changed often enough, we regularly refresh collections signatures (using [a lambda](https://github.com/mozilla-services/remote-settings-lambdas/blob/master/commands/refresh_signature.py)) if it is older than 7 days.
+
+This operation does not change the records, it only updates the signature fields. Namely, the certificate URL (x5u) and the signing value.
+
+Because of this, the CDN cache is not busted on signature refresh. This lead to signature verification errors because clients could fetch an outdated response, like pointing to an expired certificate chain.
+
+For example:
+
+1. Changes are made on collection, data timestamp is bumped (`last_modified=42`)
+1. Clients pull data from `/changeset?_expected=42`
+1. CDN caches the response
+1. Certificate expires
+1. Signature is refreshed, the collection metadata is updated, data timestamp remains the same
+1. Client queries monitor/changes endpoint, `last_modified=42`, pull changeset with `_expected=42`
+1. CDN returns the cached changeset
+1. Client pull expired certificate, signature fails
+1. Client retries, clears local data, pulls full changeset again with `_expected=42`
+1. Client signature fails again.
+
+## Decision Drivers
+
+In order to choose our solution we considered the following criteria:
+
+- **Complexity**: Low → High: how much additional complexity
+- **Impact**: Low → High: how big is the change on the whole service
+- **Efforts**: Low → High: how much efforts are necessary
+- **Rollout**: Slow → Fast: how fast can we roll out the fix
+
+## Considered Options
+
+1. Bump data timestamp on signature refresh
+1. Use metadata timestamp as cache bust
+1. Use metadata timestamp as cache bust only if signature fails
+
+## Decision Outcome
+
+Chosen option: option X because XXXX
+
+Both options 1 and 2 have a significant impact.
+
+Option 1 will cause more trafic, but globally absorbed by the CDN. It will force reverification of up-to-date data on clients, solving [bug 1640126](https://bugzilla.mozilla.org/show_bug.cgi?id=1640126). Rolling out is immediate and affect all versions.
```

Comment:
```suggestion
Option 1 will cause more traffic, but globally absorbed by the CDN. It will force reverification of up-to-date data on clients, solving [bug 1640126](https://bugzilla.mozilla.org/show_bug.cgi?id=1640126). Rolling out is immediate and affect all versions.
```